File: Easy/Java/504. Base 7.java
class Solution {
    public String convertToBase7(int num) {
        if(num == 0)
            return "0";

        String neg = 0 <= num ? "" : "-";
        String base7 = "";
        num = Math.abs(num);

        while(1 <= num){
            base7 = Integer.toString(num % 7) + base7;
            num = num / 7;
        }

        return neg + base7;
    }
}

File: Easy/Python/504. Base 7.py
class Solution(object):
    def convertToBase7(self, num):
        if num == 0:
            return '0'
            
        neg = '' if 0 <= num else '-'
        base7 = ''
        num = abs(num)

        while 1 <= num:
            base7 = str(num % 7) + base7
            num = num // 7
        
        return neg + base7
